src/Microsoft.TemplateEngine.Utils/PhysicalFileSystem.cs

```diff
@@ -44,6 +44,10 @@ public void FileCopy(string source, string target, bool overwrite)
 
         public void DirectoryDelete(string path, bool recursive)
         {
+            foreach(string file in Directory.GetFiles(path, "*", SearchOption.AllDirectories))
```

Comment:
This doesn't seem like a good thing to do, if the user marked a file as readonly or If it's marked as system, I'd say that failure is probably correct.